eng/linkerTests/linkerCorrectnessTests.targets

```diff
@@ -0,0 +1,109 @@
+      <_projectSourceFile>%(TestConsoleApps.ProjectCompileItems)</_projectSourceFile>
+    </PropertyGroup>
+
+    <MakeDir Directories="$(_projectDir)" />
+    <WriteLinesToFile File="$(_projectFile)"
+                      Lines="$([System.IO.File]::ReadAllText('$(ProjectTemplate)').Replace('{RuntimePackDir}', '%(TestConsoleApps.RuntimePackDirectory)').Replace('{TargetingPackDir}','$(MicrosoftNetCoreAppRefPackDir)').Replace('{RuntimeIdentifier}','%(TestConsoleApps.TestRuntimeIdentifier)'))"
+                      Overwrite="true" />
+    <Copy SourceFiles="$(_projectSourceFile)"
+          DestinationFolder="$(_projectDir)" />
+    <Message Text="Generated $(_projectFile)" />
+  </Target>
+
+  <Target Name="RestoreProjects"
+          DependsOnTargets="GenerateProjects">
+    <PropertyGroup>
+      <TestRestoreCommand>"$(TestDotNetPath)"</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) restore</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) /nr:false</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) /warnaserror</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) -p:configuration=Release</TestRestoreCommand>
+    </PropertyGroup>
+
+    <Exec Command="$(TestRestoreCommand)" StandardOutputImportance="Low" WorkingDirectory="%(TestConsoleApps.ProjectDir)" />
+  </Target>
+
+  <Target Name="PublishTrimmedProjects"
+          DependsOnTargets="RestoreProjects">
+    <PropertyGroup>
+      <TestRestoreCommand>"$(TestDotNetPath)"</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) msbuild /t:PublishTrimmed</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) /nr:false</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) /warnaserror</TestRestoreCommand>
+      <TestRestoreCommand>$(TestRestoreCommand) -p:configuration=Release</TestRestoreCommand>
+    </PropertyGroup>
+
+    <Exec Command="$(TestRestoreCommand)" StandardOutputImportance="Low" WorkingDirectory="%(TestConsoleApps.ProjectDir)" />
+  </Target>
+
+  <Target Name="ExecuteApplications"
+          DependsOnTargets="PublishTrimmedProjects"
+          Inputs="%(TestConsoleApps.Identity)"
+          Outputs="_unused"
+          Condition="'$(ArchiveTests)' != 'true'">
+
+    <Message Importance="High" Text="[Linker Tests] Running test: %(TestConsoleApps.ProjectCompileItems)..." />
+    <Exec IgnoreExitCode="true" Command="%(TestConsoleApps.TestCommand)" StandardOutputImportance="Low" WorkingDirectory="%(TestConsoleApps.TestExecutionDirectory)">
+      <Output TaskParameter="ExitCode" PropertyName="ExecutionExitCode" />
+    </Exec>
+
+    <Error Condition="'$(ExecutionExitCode)' != '100'" Text="Error: [Failed Test]: %(TestConsoleApps.ProjectCompileItems) The Command %(TestConsoleApps.TestCommand) return a non-success exit code." />
+  </Target>
+
+  <Target Name="Build" DependsOnTargets="ExecuteApplications" />
```

Comment:
Why do we also run the application when we build them? I assumed `dotnet build` generates the projects and `dotnet test` or `dotnet msbuild /t:Test` executes them.

Reply:
We will in the future change this once we want to start sending tests to helix, by having the ExecuteApplications target conditioned on whether or not you use ArchiveTests=true and then instead of executing we will zip and send them over to helix. This will be addressed in a subsequent PR once we need a test runtime that doesn't work on the build machines.